Modules/_io/bytesio.c

```diff
@@ -440,16 +440,14 @@ bytesio_truncate(bytesio *self, PyObject *args)
 
     CHECK_CLOSED(self);
 
-    if (PyErr_WarnPy3k_WithFix("BytesIO.truncate() does not shift the file pointer",
-                       "use seek(0) before doing truncate(0)", 1) < 0)
-    {
-        return NULL;
-    }
-
     if (!PyArg_ParseTuple(args, "|O:truncate", &arg))
         return NULL;
 
     if (PyNumber_Check(arg)) {
+        if (arg == 0 && PyErr_WarnPy3k_WithFix("BytesIO.truncate() does not shift the file pointer",
```

Comment:
I *think* `arg` has type `PyObject` so `if (arg == 0)` is really `if (arg == NULL)` which isn't the same as checking for an integer `0`. Should this `if` go after the `size = PyNumber_AsSsize_t(arg, PyExc_OverflowError);` line so that we can then say `if (size == 0) { ... }`?

Reply:
Good one, the reason I move inside here was to use the argument and number check but forgot. The new commit addresses this.